src/activities/content/ContentEntity.js

```diff
@@ -32,8 +36,13 @@ export class ContentEntity extends Entity {
 	 * @returns {string} content-module link
 	 */
 	getModuleHref() {
-		return this._entity
-			&& this._entity.hasLinkByRel(Rels.Content.moduleEntity)
-			&& this._entity.getLinkByRel(Rels.Content.moduleEntity).href;
+		return ContentHelperFunctions.getHrefFromRel(Rels.Content.moduleEntity, this._entity);
+	}
+
+	/**
+	 * @returns {string} content-weblink link
+	 */
+	getWebLinkHref() {
```

Comment:
Maybe this way with a getter for every possibility is preferred, but I'm imagining that it's not possible to have existing hrefs for different entity types. Like, if it's a module entity it will have a module href and everything else will be null. so do we need a getter for everything or do we just need one that would check the `getModuleHref` and spit out what we expect? But like I said, maybe it's preferred to have everything compartmentalized like that... But it just feels like a bunch of redundant code that will ultimately return `null` or whatever in all but 1 scenario at each time

Reply:
I guess this does prevent us from attempting to access the href for a type that doesnt exist, but that can be gotten around by making the function take in an entity type and return the href based on that type. Is that  approach going against what this class is supposed to represent and would someone be able to explain to me why?

Reply:
I see @ctwomey1  kind of alluded to this above:
>I think describing the entity fully would certainly make it more flexible for consumption. It would all you to do things like pull all activities out of a collection with class "weblink" easier, but if current workflows don't require it, i wouldn't worry about it
do you think your comment applies to what I am describing as well?

Reply:
This class (ContentEntity) represents the (somewhat) generic activity-usage object. When looking at the relevant siren response for a `module`, this is what we see:
![image](https://user-images.githubusercontent.com/2243995/103568418-1c785180-4e8b-11eb-977f-62fd11607d8c.png)

The response for a `weblink` would be pretty similar, only the classes would change, and the link to the `content-module` entity would be replaced with a link to the `content-weblink` entity.

I think ideally we would have a different class for each type, though I'm not totally sure how that would be represented in this repo. I'm also not sure if it's really worth a bunch of refactoring to achieve it, considering the push to move to `hmc-components` going forward. Since that repo does not make use of siren-sdk any work we do in here to refactor and make this representation more perfect is potentially wasted effort that could be used to get us moved over to the `hmc-components` earlier.

Reply:
Yeah I think it's fine the way it is now. Not that there can't be imporovements, but with the move to `hmc-foundation` It's not super important